truss/main.go

```diff
@@ -6,133 +6,209 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/pkg/errors"
 
-	"github.com/TuneLab/go-truss/truss/protostage"
+	"github.com/TuneLab/go-truss/truss/protostuff"
 	"github.com/TuneLab/go-truss/truss/truss"
 
 	"github.com/TuneLab/go-truss/deftree"
 	"github.com/TuneLab/go-truss/gendoc"
 	"github.com/TuneLab/go-truss/gengokit"
 )
 
-func main() {
+var (
+	pbPathFlag = flag.String("pbout", "", "The go package path where the protoc-gen-go .pb.go structs will be written.")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]... [*.proto]...\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+
 	flag.Parse()
 
 	if len(flag.Args()) == 0 {
-		exitIfError(errors.New("no arguments passed"))
+		fmt.Fprintf(os.Stderr, "%s: missing .proto file(s)\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Try '%s --help' for more information.\n", os.Args[0])
+		os.Exit(1)
 	}
+}
+
+func main() {
+	cfg, err := parseInput()
+	exitIfError(errors.Wrap(err, "could not parse input"))
+
+	dt, err := parseServiceDefinition(cfg.DefPaths)
+	exitIfError(errors.Wrap(err, "unable to parse input definition proto files"))
+
+	err = updateConfigWithService(cfg, dt)
+	exitIfError(errors.Wrap(err, "unable to read system based on service definition"))
```

Comment:
Haha I love your message because this error wraps this error: [unable to read previously generated files](https://github.com/hasAdamr/go-truss/blob/e765fa9d0e4889468a088fac5710aa19205d7383/truss/main.go#L125)

Should I wrap it again? This was a place I got stuck.